rfcs/0000-single-cluster-dnspolicy.md

```diff
@@ -0,0 +1,272 @@
+
+**Example 2.** DNSPolicy using `simple` strategy with external dns provider
+
+```yaml
+apiVersion: kuadrant.io/v1alpha1
+kind: DNSPolicy
+metadata:
+  name: prod-web
+  namespace: my-gateways
+spec:
+  providerRef:
+    name: external-dns
+    kind: ExternalDNS
+  targetRef:
+    name: prod-web
+    group: gateway.networking.k8s.io
+    kind: Gateway
+  strategy: simple
+```
+
+In ths example if the DNSPolicy was attached to the same gateway described in example 1, the same DNSRecord would also be created but MGC would not reconcile it. In this scenario it is expected that an external controller is being used to manage the reconciliation of the DNSRecord resources such as [external-dns](https://github.com/kubernetes-sigs/external-dns).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## API Updates
+
+DNSPolicy:
+
+- new providerRef field `spec.providerRef`
+- - new strategy field `spec.strategy` 
+
+DNSRecord:
+ 
+- `spec.managedZone` replaced with `spec.providerRef`
+- - new zoneID field `spec.zoneID`
+
+ManagedZone:
+
+- `spec.dnsProviderSecretRef` replaced with `spec.providerRef`
+
+### DNSPolicy.spec.providerRef
+
+The `providerRef` field is mandatory and contains a reference to a resource that shows how DNSRecords will be reconciled.
+    - `spec.providerRef.name` - name of the provider resource
+    - `spec.providerRef.kind` - kind of resource, can be anything but only `Secret` or `Managedzone` will be reconciled by MGC.
+
+A kind of type `Secret` will be managed by MGC, and a secret in the dns policies namespace with the given name must exist. The expected contents of the secrets data is comparable to the `dnsProviderSecretRef` used by ManageZones.
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: mgc-aws-credentials
+type: kuadrant.io/aws
+data:
+  AWS_ACCESS_KEY_ID: "foo"
+  AWS_SECRET_ACCESS_KEY: "bar"
+  CONFIG:
+    zoneIDFilter:
+      - Z04114632NOABXYWH93QUl
+```
+
+The `CONFIG` section of the secrets data will be added to allow provider specific configuration to be stored alongside the providers credentials and can be used during the instantiation of the provider client, and during any provider operations.
+The above for example would use the `zoneIDFilter` value to limit what hosted zones this provider is allowed to update.
+
+A kind of type `ManagedZone` will be managed by MGC, and a ManagedZone in the dns policies namespace with the given name must exist. 
+
+A kind of any other type e.g. `ExternalDNS` informs MGC that it should not reconcile any resources referencing it. All fields of `providerRef` will be ignored by MGC and can be set to anything, however since the `providerRef` will still be copied over to any DNSRecord resources created by the policy controller the values may still be given meaning to that external DNSRecord reconciler.
```

Comment:
```suggestion
A kind of any other type e.g. `ExternalDNS` informs the `DNSPolicy` controller that it should not reconcile any resources referencing it. All fields of `providerRef` will be ignored by the `DNSPolicy` controller and can be set to anything, however since the `providerRef` will still be copied over to any DNSRecord resources created by the policy controller the values may still be given meaning to that external DNSRecord reconciler.
```